src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/XmlObjectSerializerWriteContext.cs

```diff
@@ -574,7 +585,66 @@ private static void WriteIXmlSerializable(XmlWriterDelegator xmlWriter, object o
             xmlSerializableWriter.EndWrite();
         }
 
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        internal void GetObjectData(ISerializable obj, SerializationInfo serInfo, StreamingContext context)
+        {
+            obj.GetObjectData(serInfo, context);
+        }
+
+        public void WriteISerializable(XmlWriterDelegator xmlWriter, ISerializable obj)
+        {
+            Type objType = obj.GetType();
+            var serInfo = new SerializationInfo(objType, XmlObjectSerializer.FormatterConverter /*!UnsafeTypeForwardingEnabled is always false*/);
+            GetObjectData(obj, serInfo, GetStreamingContext());
+
+            if (!UnsafeTypeForwardingEnabled && serInfo.AssemblyName == Globals.MscorlibAssemblyName)
```

Comment:
There's discussion on adding NetDataContractSerializer back. I would just keep this as is.